widgets/src/main/java/org/pentaho/gwt/widgets/client/utils/NameUtils.java

```diff
@@ -76,9 +76,13 @@ public static String reservedCharListForDisplay( ) {
     return getReservedCharsDisplay();
   }
 
-  public static native String URLEncode( String value )
+  public static String URLEncode( String value ) {
```

Comment:
so public functions need to "handle" all possible arguments
` String URLEncode( String value )`

It may be valid to pass any value to the same name public native function. Just denote in javadoc that some thought has been put into handling all possible arguments. It may or may not be valid to throw an IllegalArgumentException (or another exeception) here

Reply:
this comment seems obsolete since the change of approach introduced with https://github.com/pentaho/pentaho-commons-gwt-modules/commit/f13ab321f9ecfc62868fa68c1b18a1be6bb74fee